pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/BaseSegmentFetcher.java

```diff
@@ -75,6 +77,16 @@ public void fetchSegmentToLocal(URI uri, File dest)
     });
   }
 
+  @Override
+  public void fetchSegmentToLocal(List<URI> uris, File dest)
```

Comment:
Nice.
How about changing this to apply retries with backoff but use one of the uris at random ? Sort of similar to lines 68-77